### PR TITLE
chore: Move node-core/events to standalone crate

### LINKED ISSUE #7596: extract node-core/events to standalone crate
### Describe the feature

we want to trim the node-core crate

we can extract 

https://github.com/paradigmxyz/reth/tree/dd28a4ca730f5d213861ab23ca012896c1d19c34/crates/node-core/src/events

to a standalone crate

further we can move all event types from primitives to this crate

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by supernovahs**
closes #7596

### CODE REVIEW

File: crates/events/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[package]
+name = "reth-events"
```
Comment (onbjerg) [inline crates/events/Cargo.toml:2]: i'd name the crate node-events (dir) and reth-node-events (crate)

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Review by onbjerg — APPROVED** [review]
lgtm once tests pass

**Review by mattsse — APPROVED** [review]

**Comment by supernovahs** [comment]
oops. Went away for a while

**Comment by onbjerg** [comment]
Haha, no worries! Thank you for this 😊

File: crates/node-events/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+//! Various event handlers for the node.
+
+//! Standalone crate for ethereum-specific Reth configuration and builder types.
```
Comment (onbjerg) [inline crates/node-events/src/lib.rs:3]:
```suggestion
```

rm copy pasted doc

**Pull request merged**